metrics-operator/api/v1alpha3/analysisdefinition_types.go

```diff
@@ -124,3 +124,10 @@ func init() {
 func (o *OperatorValue) GetFloatValue() float64 {
 	return o.FixedValue.AsApproximateFloat64()
 }
+
+func (o *Objective) GetAnalysisValueTemplateNamespace(namespace string) string {
+	if o.AnalysisValueTemplateRef.Namespace == "" {
```

Comment:
to reduce duplication of that logic, we can also make this a function of the `ObjectReference` struct that is used for both the AnalysisDefinition and AnalysisValueTemplate references

Reply:
you mean just reduce the logic in if statement? Something like

```
if o.AnalysisValueTemplateRef.IsNamespaceNotSet() { ...
```